qiskit_machine_learning/algorithms/classifiers/pegasos_qsvc.py

```diff
@@ -0,0 +1,285 @@
+            y: shape (n_samples), train labels . Must not contain more than two unique labels.
+            sample_weight: this parameter is not supported, passing a value raises an error.
+
+        Returns:
+            ``self``, a trained model.
+
+        Raises:
+            ValueError:
+                - X and/or y have the wrong shape.
+                - X and y have incompatible dimensions.
+                - y includes more than two unique labels.
+                - Pre-computed kernel matrix has the wrong shape and/or dimension.
+
+            NotImplementedError:
+                - when a sample_weight which is not None is passed.
+        """
+        # check whether the data have the right format
+        if np.ndim(X) != 2:
+            raise ValueError("X has to be a 2D array")
+        if np.ndim(y) != 1:
+            raise ValueError("y has to be a 1D array")
+        if len(np.unique(y)) != 2:
+            raise ValueError("Only binary classification is supported")
+        if X.shape[0] != y.shape[0]:
+            raise ValueError("'X' and 'y' have to contain the same number of samples")
+        if self._precomputed and X.shape[0] != X.shape[1]:
+            raise ValueError(
+                "For a precomputed kernel, X should be in shape (n_samples, n_samples)"
+            )
+        if sample_weight is not None:
+            raise NotImplementedError(
+                "Parameter 'sample_weight' is not supported. All samples have to be weighed equally"
+            )
+        # reset the fit state
+        self._fit_status = False
+
+        # the algorithm works with labels in {+1, -1}
+        self._label_pos = np.unique(y)[0]
+        self._label_neg = np.unique(y)[1]
+        self._label_map = {self._label_pos: +1, self._label_neg: -1}
+
+        # the training data are later needed for prediction
+        self._x_train = X
+        self._y_train = y
+        self._n_samples = X.shape[0]
+
+        # empty dictionary to represent sparse array
+        self._alphas = {}
+
+        t_0 = datetime.now()
+        # training loop
+        for step in range(1, self._num_steps + 1):
+            # for every step, a random index (determining a random datum) is fixed
+            i = np.random.randint(0, len(y))
+
+            value = self._compute_weighted_kernel_sum(i, X, training=True)
+
+            if (self._label_map[y[i]] * self.C / step) * value < 1:
+                # only way for a component of alpha to become non zero
+                self._alphas[i] = self._alphas.get(i, 0) + 1
+
+        self._fit_status = True
+
+        logger.debug("fit completed after %s", str(datetime.now() - t_0)[:-7])
+
+        return self
+
+    # pylint: disable=invalid-name
+    def predict(self, X: np.ndarray) -> np.ndarray:
+        """
+        Perform classification on samples in X.
+
+        Args:
+            X: Features. For a callable kernel shape (m_samples, n_features), for a precomputed
```

Comment:
Same comment as in 'fit' about callable kernel

Reply:
Updated in the same manner.